src/lsm/tree_fuzz.zig

```diff
@@ -0,0 +1,432 @@
+const std = @import("std");
+const testing = std.testing;
+const allocator = testing.allocator;
+const assert = std.debug.assert;
+
+const config = @import("../config.zig");
+const fuzz = @import("../test/fuzz.zig");
+const vsr = @import("../vsr.zig");
+const log = std.log.scoped(.lsm_forest_fuzz);
+
+const MessagePool = @import("../message_pool.zig").MessagePool;
+const Transfer = @import("../tigerbeetle.zig").Transfer;
+const Account = @import("../tigerbeetle.zig").Account;
+const Storage = @import("../test/storage.zig").Storage;
+const StateMachine = @import("../state_machine.zig").StateMachineType(Storage);
+
+const GridType = @import("grid.zig").GridType;
+const NodePool = @import("node_pool.zig").NodePool(config.lsm_manifest_node_size, 16);
+const Table = @import("table.zig").TableType(
+    Key,
+    Value,
+    compare_keys,
+    key_from_value,
+    sentinel_key,
+    tombstone,
+    tombstone_from_key,
+);
+const Tree = @import("tree.zig").TreeType(Table, Storage, @typeName(Table) ++ "_test");
+
+const Grid = GridType(Storage);
+const SuperBlock = vsr.SuperBlockType(Storage);
+
+const Key = packed struct {
+    id: u64 align(@alignOf(u64)),
+};
```

Comment:
It has to be packed or extern to guarantee the representation on disk. It fails to compile otherwise.

The align is needed to satisfy https://github.com/tigerbeetledb/tigerbeetle/blob/main/src/lsm/table.zig#L106. Without the annotation the alignment is 1. I don't know why Key needs the annotation but Value doesn't.

Reply:
Gotcha, thanks. I'm not sure about the Value alignment thing either.